controller/src/main/java/io/pravega/controller/store/host/ZKHostStore.java

```diff
@@ -83,7 +83,7 @@ private void tryInit() {
             ZKUtils.createPathIfNotExists(zkClient, zkPath, HostContainerMap.EMPTY.toBytes());
             hostContainerMapNode.getListenable().addListener(this::updateMap);
             hostContainerMapNode.start(true);
-
+            updateMap();
```

Comment:
My understanding is that we trigger this method upon an update by registering it with  `addListener` in line 84. But, given that there might not be a change for an arbitrary amount of time, we need to trigger it manually the first time. I can't determine whether this initial call needs to happen before line 83 or after line 85 like in this PR, or doesn't matter.

Reply:
> I can't determine whether this initial call needs to happen before line 83 or after line 85 like in this PR, or doesn't matter.

It doesnt matter because `initialize` and `updateMap` are both synchronized. So even if an update was triggered from the listener, the first call will be from the init itself as it already has the lock..

Reply:
the alternate option was to not build initial cache.. 
```
             hostContainerMapNode.start(false); 
```
this would mean it will read the value in the background and since a listener is registered, updatemap will be called from the listener when the value is updated. 
Both approaches will have the same effect.